src/components/Carousel.js

```diff
@@ -0,0 +1,113 @@
+import mjml2html from 'mjml';
+import { isComponentType } from '.';
+
+export default (editor, { dc, coreMjmlModel, coreMjmlView, sandboxEl }) => {
+  const type = 'mj-carousel';
+
+  dc.addType(type, {
+    isComponent: isComponentType(type),
+    model: {
+      ...coreMjmlModel,
+      defaults: {
+        name: 'Carousel',
+        draggable: '[data-gjs-type=mj-column]',
+        droppable: '[data-gjs-type=mj-carousel-image]',
+        'style-default': {
+          // TODO
+        },
+        stylable: [
+          // TODO
+        ],
+      },
+    },
+
+    view: {
+      ...coreMjmlView,
+
+      tagName: 'tr',
+
+      attributes: {
+        style: 'pointer-events: all; display: table; width: 100%',
+      },
+
+      init() {
+        coreMjmlView.init.call(this);
+        this.listenTo(this.model.get('components'), 'add remove', this.render);
+      },
+
+      getTemplateFromMjml() {
+        let mjmlTmpl = this.getMjmlTemplate();
+        let innerMjml = this.getInnerMjmlTemplate();
+        const htmlOutput = mjml2html(`${mjmlTmpl.start}
+          ${innerMjml.start}${innerMjml.end}${mjmlTmpl.end}`);
+        let html = htmlOutput.html;
+
+        let styles = [];
+        sandboxEl.innerHTML = html;
+        var styleArr = Array.from(sandboxEl.querySelectorAll('style'));
+        styleArr.forEach((item) => {
+          styles.push(item.innerHTML);
+        });
+
+
+        let content = html.replace(/<body(.*)>/, '<body>');
+        let start = content.indexOf('<body>') + 6;
+        let end = content.indexOf('</body>');
+        sandboxEl.innerHTML = content.substring(start, end).trim();
+        let componentEl = this.getTemplateFromEl(sandboxEl);
+
+        // Copy all rendered attributes (TODO need for all)
+        let attributes = {};
+        const elAttrs = componentEl.attributes;
+
+        for (let elAttr, i = 0, len = elAttrs.length; i < len; i++) {
+          elAttr = elAttrs[i];
+          attributes[elAttr.name] = elAttr.value;
+        }
+
+        return {
+          attributes,
+          content: componentEl.innerHTML,
+          style: styles.join(' ')
+        };
+      },
```

Comment:
I think `getTemplateFromMjml` code is basically a copy/paste from the already used Column and NavBar. Probably it would make sense to abstract it and to be DRY as much as possible.

Unfortunately, I didn't yet start working on API for creating custom renderer but once it will be ready it will be easier to switch if we have less code